### DISCUSSION TITLE
sliding_window_inferer use case

### DISCUSSION THREAD

**Discussion by justinbroce** [Project-MONAI/MONAI#6035] [General]
I'm working a project using swinUNETR for 2D segmentation, and am curious what purpose sliding_window_inference serves that the forward function does not. are the two interchangeable. is there a reason that i should use one over the other? thanks

**Comment by KumoLiu** [comment]
Hi @justinbroce, do you mean why we do `sliding_window_inference` during validation and test, but not do it during training? First, you could also do `sliding_window_inference` during training. The reason why we don't do it during training is that:

- First, as long as the number of epochs trained is large enough, we can iterate all of the training data at a faster rate.
- Second, we could also save memory when the data is too large for training.

Hope it can help you, thanks!

**Comment by tangy5** [comment]
Sliding_window_inference performs the prediction upon all patches (sliding windows) by the given image volume, then we can get a whole inferenced prediction. The forward function in the network output each patch prediction. They are doing different jobs, Sling_window_inference uses the network/model for inference.

**Reply by justinbroce** [reply]
Thanks for the explanation. I didn't mean to say forward. I am more asking about the difference between using the model itself for inference vs using sliding_window_inference.
